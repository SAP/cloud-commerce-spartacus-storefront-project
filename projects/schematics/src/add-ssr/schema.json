{
  "$schema": "http://json-schema.org/schema",
  "id": "SpartacusSchematics",
  "title": "Spartacus Schematics",
  "type": "object",
  "properties": {
    "project": {
      "type": "string",
      "description": "The name of the project.",
      "$default": {
        "$source": "projectName"
      }
    }
  },
  "required": []
}
