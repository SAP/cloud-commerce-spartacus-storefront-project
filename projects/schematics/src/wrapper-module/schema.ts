import { ExecutionOptions } from '@angular-devkit/schematics';

export interface Schema extends Partial<ExecutionOptions> {
  /**
   * The name of the project in which to execute schematics.
   */
  project: string;
  /**
   * When enabled, prints the additional logs.
   */
  debug?: boolean;
  /**
   * The marker module for which
   * to find or generate a new wrapper module.
   */
  markerModuleName: string;
  /**
   * Name of the feature module
   * to add to the wrapper module.
   */
  featureModuleName: string;
  /**
   * Internal options.
   * Should not be set by the user.
   */
  internal?: {
    /**
     * The execution sequence.
     * Used for ensuring one log message is printed.
     */
    sequence?: number;
  };
}
