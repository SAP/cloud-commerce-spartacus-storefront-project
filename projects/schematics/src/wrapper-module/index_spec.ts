import { Tree } from '@angular-devkit/schematics';
import { SchematicTestRunner } from '@angular-devkit/schematics/testing';
import {
  Schema as ApplicationOptions,
  Style,
} from '@schematics/angular/application/schema';
import { Schema as WorkspaceOptions } from '@schematics/angular/workspace/schema';
import * as path from 'path';
import { Schema as SpartacusOptions } from '../add-spartacus/schema';
import { CART_BASE_MODULE } from '../shared/lib-configs/cart-schematics-config';
import { CHECKOUT_BASE_MODULE } from '../shared/lib-configs/checkout-schematics-config';
import {
  CART_BASE_FEATURE_NAME,
  CDC_FEATURE_NAME,
  CHECKOUT_B2B_FEATURE_NAME,
  CHECKOUT_BASE_FEATURE_NAME,
  CHECKOUT_SCHEDULED_REPLENISHMENT_FEATURE_NAME,
  DIGITAL_PAYMENTS_FEATURE_NAME,
  SPARTACUS_SCHEMATICS,
} from '../shared/libs-constants';
import { LibraryOptions } from '../shared/utils/lib-utils';
import { createProgram } from '../shared/utils/program';
import { getProjectTsConfigPaths } from '../shared/utils/project-tsconfig-paths';
import {
  cartBaseFeatureModulePath,
  cdcFeatureModulePath,
  checkoutFeatureModulePath,
  checkoutWrapperModulePath,
  digitalPaymentsFeatureModulePath,
  spartacusFeaturesModulePath,
  userFeatureModulePath,
  userWrapperModulePath,
} from '../shared/utils/test-utils';
import { Schema as SpartacusWrapperOptions } from '../wrapper-module/schema';

const collectionPath = path.join(__dirname, '../collection.json');

describe('Spartacus Wrapper Module Schematics: ng g @spartacus/schematics:wrapper-module', () => {
  const schematicRunner = new SchematicTestRunner(
    SPARTACUS_SCHEMATICS,
    collectionPath
  );

  let appTree: Tree;
  let buildPath: string;

  const workspaceOptions: WorkspaceOptions = {
    name: 'workspace',
    version: '0.5.0',
  };

  const appOptions: ApplicationOptions = {
    name: 'schematics-test',
    inlineStyle: false,
    inlineTemplate: false,
    routing: false,
    style: Style.Scss,
    skipTests: false,
    projectRoot: '',
  };

  const defaultOptions: SpartacusOptions = {
    project: 'schematics-test',
    lazy: true,
    features: [],
  };

  const BASE_OPTIONS: LibraryOptions = {
    project: 'schematics-test',
    lazy: true,
  };

  beforeEach(async () => {
    appTree = await schematicRunner
      .runExternalSchematicAsync(
        '@schematics/angular',
        'workspace',
        workspaceOptions
      )
      .toPromise();
    appTree = await schematicRunner
      .runExternalSchematicAsync(
        '@schematics/angular',
        'application',
        appOptions,
        appTree
      )
      .toPromise();

    buildPath = getProjectTsConfigPaths(appTree, BASE_OPTIONS.project)
      .buildPaths[0];
  });

  describe('One dynamic import in the file', () => {
    it('should generate appropriate feature module', async () => {
      appTree = await schematicRunner
        .runSchematicAsync(
          'ng-add',
          {
            ...defaultOptions,
            features: [CHECKOUT_B2B_FEATURE_NAME],
            name: 'schematics-test',
          },
          appTree
        )
        .toPromise();

      const { program } = createProgram(appTree, appTree.root.path, buildPath);

      const checkoutFeatureModule = program.getSourceFileOrThrow(
        checkoutFeatureModulePath
      );
      const checkoutWrapperModule = program.getSourceFileOrThrow(
        checkoutWrapperModulePath
      );

      expect(checkoutFeatureModule.print()).toMatchSnapshot();
      expect(checkoutWrapperModule.print()).toMatchSnapshot();
    });
  });

  describe('Multiple dynamic imports in the file', () => {
    it('should generate appropriate feature module', async () => {
      appTree = await schematicRunner
        .runSchematicAsync(
          'ng-add',
          {
            ...defaultOptions,
            features: [CART_BASE_FEATURE_NAME],
            name: 'schematics-test',
          },
          appTree
        )
        .toPromise();
      const options: SpartacusWrapperOptions = {
        project: 'schematics-test',
        markerModuleName: CART_BASE_MODULE,
        featureModuleName: CART_BASE_MODULE,
      };
      appTree = await schematicRunner
        .runSchematicAsync('wrapper-module', options, appTree)
        .toPromise();

      const { program } = createProgram(appTree, appTree.root.path, buildPath);

      const cartFeatureModule = program.getSourceFileOrThrow(
        cartBaseFeatureModulePath
      );
      const cartWrapperModule = program.getSourceFileOrThrow(
        'src/app/spartacus/features/cart/cart-base-wrapper.module.ts'
      );

      expect(cartFeatureModule.print()).toMatchSnapshot();
      expect(cartWrapperModule.print()).toMatchSnapshot();
    });
  });

  describe('Double execution', () => {
    it('should not change anything', async () => {
      // first execution happens under the hood
      appTree = await schematicRunner
        .runSchematicAsync(
          'ng-add',
          {
            ...defaultOptions,
            features: [CHECKOUT_BASE_FEATURE_NAME],
            name: 'schematics-test',
          },
          appTree
        )
        .toPromise();

      const options: SpartacusWrapperOptions = {
        project: 'schematics-test',
        markerModuleName: CHECKOUT_BASE_MODULE,
        featureModuleName: CHECKOUT_BASE_MODULE,
      };
      // the second execution
      appTree = await schematicRunner
        .runSchematicAsync('wrapper-module', options, appTree)
        .toPromise();

      const { program } = createProgram(appTree, appTree.root.path, buildPath);

      const checkoutFeatureModule = program.getSourceFileOrThrow(
        checkoutFeatureModulePath
      );
      const checkoutWrapperModule = program.getSourceFileOrThrow(
        checkoutWrapperModulePath
      );

      expect(checkoutFeatureModule.print()).toMatchSnapshot();
      expect(checkoutWrapperModule.print()).toMatchSnapshot();
    });
  });

  describe('Checkout Scheduled Replenishment', () => {
    it('should create the checkout wrapper module and import Checkout features', async () => {
      appTree = await schematicRunner
        .runSchematicAsync(
          'ng-add',
          {
            ...defaultOptions,
            name: 'schematics-test',
            features: [CHECKOUT_SCHEDULED_REPLENISHMENT_FEATURE_NAME],
          },
          appTree
        )
        .toPromise();

      const { program } = createProgram(appTree, appTree.root.path, buildPath);

      const checkoutFeatureModule = program.getSourceFileOrThrow(
        checkoutFeatureModulePath
      );
      const checkoutWrapperModule = program.getSourceFileOrThrow(
        checkoutWrapperModulePath
      );

      expect(checkoutFeatureModule.print()).toMatchSnapshot();
      expect(checkoutWrapperModule.print()).toMatchSnapshot();
    });
  });

  describe('Digital Payments', () => {
    it('should create the checkout wrapper module and import Base Checkout and DP', async () => {
      appTree = await schematicRunner
        .runSchematicAsync(
          'ng-add',
          {
            ...defaultOptions,
            name: 'schematics-test',
            features: [DIGITAL_PAYMENTS_FEATURE_NAME],
          },
          appTree
        )
        .toPromise();

      const { program } = createProgram(appTree, appTree.root.path, buildPath);

      const checkoutWrapperModule = program.getSourceFileOrThrow(
        checkoutWrapperModulePath
      );
      const checkoutFeatureModule = program.getSourceFileOrThrow(
        checkoutFeatureModulePath
      );
      const dpFeaturesModule = program.getSourceFileOrThrow(
        digitalPaymentsFeatureModulePath
      );

      expect(checkoutWrapperModule.print()).toMatchSnapshot();
      expect(checkoutFeatureModule.print()).toMatchSnapshot();
      expect(dpFeaturesModule.print()).toMatchSnapshot();
    });
  });

  describe('CDC', () => {
    it('should create the User wrapper module and import User Profile and CDC', async () => {
      appTree = await schematicRunner
        .runSchematicAsync(
          'ng-add',
          {
            ...defaultOptions,
            name: 'schematics-test',
            features: [CDC_FEATURE_NAME],
          },
          appTree
        )
        .toPromise();

      const { program } = createProgram(appTree, appTree.root.path, buildPath);

      const userWrapperModule = program.getSourceFileOrThrow(
        userWrapperModulePath
      );
      const userFeatureModule = program.getSourceFileOrThrow(
        userFeatureModulePath
      );
      const cdcFeaturesModule =
        program.getSourceFileOrThrow(cdcFeatureModulePath);

      expect(userWrapperModule.print()).toMatchSnapshot();
      expect(userFeatureModule.print()).toMatchSnapshot();
      expect(cdcFeaturesModule.print()).toMatchSnapshot();
    });
  });

  describe('Checkout and DP', () => {
    it('Should order the imports in the wrapper and Spartacus features modules', async () => {
      appTree = await schematicRunner
        .runSchematicAsync(
          'ng-add',
          {
            ...defaultOptions,
            name: 'schematics-test',
            features: [
              CHECKOUT_SCHEDULED_REPLENISHMENT_FEATURE_NAME,
              DIGITAL_PAYMENTS_FEATURE_NAME,
            ],
          },
          appTree
        )
        .toPromise();

      const { program } = createProgram(appTree, appTree.root.path, buildPath);

      const spartacusFeaturesModule = program.getSourceFileOrThrow(
        spartacusFeaturesModulePath
      );
      const checkoutFeatureModule = program.getSourceFileOrThrow(
        checkoutFeatureModulePath
      );
      const checkoutWrapperModule = program.getSourceFileOrThrow(
        checkoutWrapperModulePath
      );
      const dpFeaturesModule = program.getSourceFileOrThrow(
        digitalPaymentsFeatureModulePath
      );
      expect(spartacusFeaturesModule.print()).toMatchSnapshot();
      expect(checkoutFeatureModule.print()).toMatchSnapshot();
      expect(checkoutWrapperModule.print()).toMatchSnapshot();
      expect(dpFeaturesModule.print()).toMatchSnapshot();
    });
  });
});
