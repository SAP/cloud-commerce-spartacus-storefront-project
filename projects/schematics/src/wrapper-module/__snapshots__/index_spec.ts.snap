// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spartacus Wrapper Module Schematics: ng g @commerce-storefront-toolset/schematics:wrapper-module Checkout Scheduled Replenishment should create the checkout wrapper module and import Checkout features 1`] = `
"import { NgModule } from '@angular/core';
import { checkoutB2BTranslationChunksConfig, checkoutB2BTranslations } from \\"@commerce-storefront-toolset/checkout/b2b/assets\\";
import { CheckoutB2BRootModule } from \\"@commerce-storefront-toolset/checkout/b2b/root\\";
import { checkoutTranslationChunksConfig, checkoutTranslations } from \\"@commerce-storefront-toolset/checkout/base/assets\\";
import { CheckoutRootModule, CHECKOUT_FEATURE } from \\"@commerce-storefront-toolset/checkout/base/root\\";
import { checkoutScheduledReplenishmentTranslationChunksConfig, checkoutScheduledReplenishmentTranslations } from \\"@commerce-storefront-toolset/checkout/scheduled-replenishment/assets\\";
import { CheckoutScheduledReplenishmentRootModule } from \\"@commerce-storefront-toolset/checkout/scheduled-replenishment/root\\";
import { CmsConfig, I18nConfig, provideConfig } from \\"@commerce-storefront-toolset/core\\";
@NgModule({
    declarations: [],
    imports: [
        CheckoutRootModule,
        CheckoutB2BRootModule,
        CheckoutScheduledReplenishmentRootModule
    ],
    providers: [provideConfig(<CmsConfig>{
            featureModules: {
                [CHECKOUT_FEATURE]: {
                    module: () => import('./checkout-wrapper.module').then((m) => m.CheckoutWrapperModule),
                },
            }
        }),
        provideConfig(<I18nConfig>{
            i18n: {
                resources: checkoutTranslations,
                chunks: checkoutTranslationChunksConfig,
            },
        }),
        provideConfig(<I18nConfig>{
            i18n: {
                resources: checkoutB2BTranslations,
                chunks: checkoutB2BTranslationChunksConfig,
            },
        }),
        provideConfig(<I18nConfig>{
            i18n: {
                resources: checkoutScheduledReplenishmentTranslations,
                chunks: checkoutScheduledReplenishmentTranslationChunksConfig,
            },
        })
    ]
})
export class CheckoutFeatureModule {
}
"
`;

exports[`Spartacus Wrapper Module Schematics: ng g @commerce-storefront-toolset/schematics:wrapper-module Checkout Scheduled Replenishment should create the checkout wrapper module and import Checkout features 2`] = `
"import { NgModule } from '@angular/core';
import { CheckoutB2BModule } from \\"@commerce-storefront-toolset/checkout/b2b\\";
import { CheckoutModule } from \\"@commerce-storefront-toolset/checkout/base\\";
import { CheckoutScheduledReplenishmentModule } from \\"@commerce-storefront-toolset/checkout/scheduled-replenishment\\";
@NgModule({
    declarations: [],
    imports: [
        CheckoutModule,
        CheckoutB2BModule,
        CheckoutScheduledReplenishmentModule
    ]
})
export class CheckoutWrapperModule {
}
"
`;

exports[`Spartacus Wrapper Module Schematics: ng g @commerce-storefront-toolset/schematics:wrapper-module Checkout and DP Should order the imports in the wrapper and Spartacus features modules 1`] = `
"import { NgModule } from '@angular/core';
import { AnonymousConsentsModule, AuthModule, CostCenterOccModule, ExternalRoutesModule, ProductModule, ProductOccModule, UserModule, UserOccModule } from \\"@commerce-storefront-toolset/core\\";
import { AddressBookModule, AnonymousConsentManagementBannerModule, AnonymousConsentsDialogModule, BannerCarouselModule, BannerModule, BreadcrumbModule, CategoryNavigationModule, CmsParagraphModule, ConsentManagementModule, FooterNavigationModule, HamburgerMenuModule, HomePageEventModule, LinkModule, LoginRouteModule, LogoutModule, MyCouponsModule, MyInterestsModule, NavigationEventModule, NavigationModule, NotificationPreferenceModule, PageTitleModule, PaymentMethodsModule, ProductCarouselModule, ProductDetailsPageModule, ProductFacetNavigationModule, ProductImagesModule, ProductIntroModule, ProductListingPageModule, ProductListModule, ProductPageEventModule, ProductReferencesModule, ProductSummaryModule, ProductTabsModule, ScrollToTopModule, SearchBoxModule, SiteContextSelectorModule, StockNotificationModule, TabParagraphContainerModule } from \\"@commerce-storefront-toolset/storefront\\";
import { UserFeatureModule } from './features/user/user-feature.module';
import { CartBaseFeatureModule } from './features/cart/cart-base-feature.module';
import { OrderFeatureModule } from './features/order/order-feature.module';
import { CheckoutFeatureModule } from './features/checkout/checkout-feature.module';
import { DigitalPaymentsFeatureModule } from './features/digital-payments/digital-payments-feature.module';
@NgModule({
    declarations: [],
    imports: [
        AuthModule.forRoot(),
        LogoutModule,
        LoginRouteModule,
        HamburgerMenuModule,
        SiteContextSelectorModule,
        LinkModule,
        BannerModule,
        CmsParagraphModule,
        TabParagraphContainerModule,
        BannerCarouselModule,
        CategoryNavigationModule,
        NavigationModule,
        FooterNavigationModule,
        BreadcrumbModule,
        ScrollToTopModule,
        PageTitleModule,
        UserModule,
        UserOccModule,
        AddressBookModule,
        PaymentMethodsModule,
        NotificationPreferenceModule,
        MyInterestsModule,
        StockNotificationModule,
        ConsentManagementModule,
        MyCouponsModule,
        AnonymousConsentsModule.forRoot(),
        AnonymousConsentsDialogModule,
        AnonymousConsentManagementBannerModule,
        ProductModule.forRoot(),
        ProductOccModule,
        ProductDetailsPageModule,
        ProductListingPageModule,
        ProductListModule,
        SearchBoxModule,
        ProductFacetNavigationModule,
        ProductTabsModule,
        ProductCarouselModule,
        ProductReferencesModule,
        ProductImagesModule,
        ProductSummaryModule,
        ProductIntroModule,
        CostCenterOccModule,
        NavigationEventModule,
        HomePageEventModule,
        ProductPageEventModule,
        ExternalRoutesModule.forRoot(),
        UserFeatureModule,
        CartBaseFeatureModule,
        OrderFeatureModule,
        CheckoutFeatureModule,
        DigitalPaymentsFeatureModule
    ]
})
export class SpartacusFeaturesModule {
}
"
`;

exports[`Spartacus Wrapper Module Schematics: ng g @commerce-storefront-toolset/schematics:wrapper-module Checkout and DP Should order the imports in the wrapper and Spartacus features modules 2`] = `
"import { NgModule } from '@angular/core';
import { checkoutB2BTranslationChunksConfig, checkoutB2BTranslations } from \\"@commerce-storefront-toolset/checkout/b2b/assets\\";
import { CheckoutB2BRootModule } from \\"@commerce-storefront-toolset/checkout/b2b/root\\";
import { checkoutTranslationChunksConfig, checkoutTranslations } from \\"@commerce-storefront-toolset/checkout/base/assets\\";
import { CheckoutRootModule, CHECKOUT_FEATURE } from \\"@commerce-storefront-toolset/checkout/base/root\\";
import { checkoutScheduledReplenishmentTranslationChunksConfig, checkoutScheduledReplenishmentTranslations } from \\"@commerce-storefront-toolset/checkout/scheduled-replenishment/assets\\";
import { CheckoutScheduledReplenishmentRootModule } from \\"@commerce-storefront-toolset/checkout/scheduled-replenishment/root\\";
import { CmsConfig, I18nConfig, provideConfig } from \\"@commerce-storefront-toolset/core\\";
@NgModule({
    declarations: [],
    imports: [
        CheckoutRootModule,
        CheckoutB2BRootModule,
        CheckoutScheduledReplenishmentRootModule
    ],
    providers: [provideConfig(<CmsConfig>{
            featureModules: {
                [CHECKOUT_FEATURE]: {
                    module: () => import('./checkout-wrapper.module').then((m) => m.CheckoutWrapperModule),
                },
            }
        }),
        provideConfig(<I18nConfig>{
            i18n: {
                resources: checkoutTranslations,
                chunks: checkoutTranslationChunksConfig,
            },
        }),
        provideConfig(<I18nConfig>{
            i18n: {
                resources: checkoutB2BTranslations,
                chunks: checkoutB2BTranslationChunksConfig,
            },
        }),
        provideConfig(<I18nConfig>{
            i18n: {
                resources: checkoutScheduledReplenishmentTranslations,
                chunks: checkoutScheduledReplenishmentTranslationChunksConfig,
            },
        })
    ]
})
export class CheckoutFeatureModule {
}
"
`;

exports[`Spartacus Wrapper Module Schematics: ng g @commerce-storefront-toolset/schematics:wrapper-module Checkout and DP Should order the imports in the wrapper and Spartacus features modules 3`] = `
"import { NgModule } from '@angular/core';
import { CheckoutB2BModule } from \\"@commerce-storefront-toolset/checkout/b2b\\";
import { CheckoutModule } from \\"@commerce-storefront-toolset/checkout/base\\";
import { CheckoutScheduledReplenishmentModule } from \\"@commerce-storefront-toolset/checkout/scheduled-replenishment\\";
import { DigitalPaymentsModule } from \\"@commerce-storefront-toolset/digital-payments\\";
@NgModule({
    declarations: [],
    imports: [
        CheckoutModule,
        CheckoutB2BModule,
        CheckoutScheduledReplenishmentModule,
        DigitalPaymentsModule
    ]
})
export class CheckoutWrapperModule {
}
"
`;

exports[`Spartacus Wrapper Module Schematics: ng g @commerce-storefront-toolset/schematics:wrapper-module Checkout and DP Should order the imports in the wrapper and Spartacus features modules 4`] = `
"import { NgModule } from '@angular/core';
import { I18nConfig, provideConfig } from \\"@commerce-storefront-toolset/core\\";
import { dpTranslationChunksConfig, dpTranslations } from \\"@commerce-storefront-toolset/digital-payments/assets\\";
@NgModule({
    declarations: [],
    imports: [],
    providers: [provideConfig(<I18nConfig>{
            i18n: {
                resources: dpTranslations,
                chunks: dpTranslationChunksConfig,
            },
        })
    ]
})
export class DigitalPaymentsFeatureModule {
}
"
`;

exports[`Spartacus Wrapper Module Schematics: ng g @commerce-storefront-toolset/schematics:wrapper-module Digital Payments should create the checkout wrapper module and import Base Checkout and DP 1`] = `
"import { NgModule } from '@angular/core';
import { CheckoutModule } from \\"@commerce-storefront-toolset/checkout/base\\";
import { DigitalPaymentsModule } from \\"@commerce-storefront-toolset/digital-payments\\";
@NgModule({
    declarations: [],
    imports: [
        CheckoutModule,
        DigitalPaymentsModule
    ]
})
export class CheckoutWrapperModule {
}
"
`;

exports[`Spartacus Wrapper Module Schematics: ng g @commerce-storefront-toolset/schematics:wrapper-module Digital Payments should create the checkout wrapper module and import Base Checkout and DP 2`] = `
"import { NgModule } from '@angular/core';
import { checkoutTranslationChunksConfig, checkoutTranslations } from \\"@commerce-storefront-toolset/checkout/base/assets\\";
import { CheckoutRootModule, CHECKOUT_FEATURE } from \\"@commerce-storefront-toolset/checkout/base/root\\";
import { CmsConfig, I18nConfig, provideConfig } from \\"@commerce-storefront-toolset/core\\";
@NgModule({
    declarations: [],
    imports: [
        CheckoutRootModule
    ],
    providers: [provideConfig(<CmsConfig>{
            featureModules: {
                [CHECKOUT_FEATURE]: {
                    module: () => import('./checkout-wrapper.module').then((m) => m.CheckoutWrapperModule),
                },
            }
        }),
        provideConfig(<I18nConfig>{
            i18n: {
                resources: checkoutTranslations,
                chunks: checkoutTranslationChunksConfig,
            },
        })
    ]
})
export class CheckoutFeatureModule {
}
"
`;

exports[`Spartacus Wrapper Module Schematics: ng g @commerce-storefront-toolset/schematics:wrapper-module Digital Payments should create the checkout wrapper module and import Base Checkout and DP 3`] = `
"import { NgModule } from '@angular/core';
import { I18nConfig, provideConfig } from \\"@commerce-storefront-toolset/core\\";
import { dpTranslationChunksConfig, dpTranslations } from \\"@commerce-storefront-toolset/digital-payments/assets\\";
@NgModule({
    declarations: [],
    imports: [],
    providers: [provideConfig(<I18nConfig>{
            i18n: {
                resources: dpTranslations,
                chunks: dpTranslationChunksConfig,
            },
        })
    ]
})
export class DigitalPaymentsFeatureModule {
}
"
`;

exports[`Spartacus Wrapper Module Schematics: ng g @commerce-storefront-toolset/schematics:wrapper-module Double execution should not change anything 1`] = `
"import { NgModule } from '@angular/core';
import { checkoutTranslationChunksConfig, checkoutTranslations } from \\"@commerce-storefront-toolset/checkout/base/assets\\";
import { CheckoutRootModule, CHECKOUT_FEATURE } from \\"@commerce-storefront-toolset/checkout/base/root\\";
import { CmsConfig, I18nConfig, provideConfig } from \\"@commerce-storefront-toolset/core\\";
@NgModule({
    declarations: [],
    imports: [
        CheckoutRootModule
    ],
    providers: [provideConfig(<CmsConfig>{
            featureModules: {
                [CHECKOUT_FEATURE]: {
                    module: () => import('./checkout-wrapper.module').then((m) => m.CheckoutWrapperModule),
                },
            }
        }),
        provideConfig(<I18nConfig>{
            i18n: {
                resources: checkoutTranslations,
                chunks: checkoutTranslationChunksConfig,
            },
        })
    ]
})
export class CheckoutFeatureModule {
}
"
`;

exports[`Spartacus Wrapper Module Schematics: ng g @commerce-storefront-toolset/schematics:wrapper-module Double execution should not change anything 2`] = `
"import { NgModule } from '@angular/core';
import { CheckoutModule } from \\"@commerce-storefront-toolset/checkout/base\\";
@NgModule({
    declarations: [],
    imports: [
        CheckoutModule
    ]
})
export class CheckoutWrapperModule {
}
"
`;

exports[`Spartacus Wrapper Module Schematics: ng g @commerce-storefront-toolset/schematics:wrapper-module Multiple dynamic imports in the file should generate appropriate feature module 1`] = `
"import { NgModule } from '@angular/core';
import { cartBaseTranslationChunksConfig, cartBaseTranslations } from \\"@commerce-storefront-toolset/cart/base/assets\\";
import { ADD_TO_CART_FEATURE, CartBaseRootModule, CART_BASE_FEATURE, MINI_CART_FEATURE } from \\"@commerce-storefront-toolset/cart/base/root\\";
import { CmsConfig, I18nConfig, provideConfig } from \\"@commerce-storefront-toolset/core\\";
@NgModule({
    declarations: [],
    imports: [
        CartBaseRootModule
    ],
    providers: [provideConfig(<CmsConfig>{
            featureModules: {
                [CART_BASE_FEATURE]: {
                    module: () => import('./cart-base-wrapper.module').then((m) => m.CartBaseWrapperModule),
                },
            }
        }),
        provideConfig(<CmsConfig>{
            featureModules: {
                [MINI_CART_FEATURE]: {
                    module: () => import('@commerce-storefront-toolset/cart/base/components/mini-cart').then((m) => m.MiniCartModule),
                },
            }
        }),
        provideConfig(<CmsConfig>{
            featureModules: {
                [ADD_TO_CART_FEATURE]: {
                    module: () => import('@commerce-storefront-toolset/cart/base/components/add-to-cart').then((m) => m.AddToCartModule),
                },
            }
        }),
        provideConfig(<I18nConfig>{
            i18n: {
                resources: cartBaseTranslations,
                chunks: cartBaseTranslationChunksConfig,
            },
        })
    ]
})
export class CartBaseFeatureModule {
}
"
`;

exports[`Spartacus Wrapper Module Schematics: ng g @commerce-storefront-toolset/schematics:wrapper-module Multiple dynamic imports in the file should generate appropriate feature module 2`] = `
"import { NgModule } from '@angular/core';
import { CartBaseModule } from \\"@commerce-storefront-toolset/cart/base\\";
@NgModule({
    declarations: [],
    imports: [
        CartBaseModule
    ]
})
export class CartBaseWrapperModule {
}
"
`;

exports[`Spartacus Wrapper Module Schematics: ng g @commerce-storefront-toolset/schematics:wrapper-module One dynamic import in the file should generate appropriate feature module 1`] = `
"import { NgModule } from '@angular/core';
import { checkoutB2BTranslationChunksConfig, checkoutB2BTranslations } from \\"@commerce-storefront-toolset/checkout/b2b/assets\\";
import { CheckoutB2BRootModule } from \\"@commerce-storefront-toolset/checkout/b2b/root\\";
import { checkoutTranslationChunksConfig, checkoutTranslations } from \\"@commerce-storefront-toolset/checkout/base/assets\\";
import { CheckoutRootModule, CHECKOUT_FEATURE } from \\"@commerce-storefront-toolset/checkout/base/root\\";
import { CmsConfig, I18nConfig, provideConfig } from \\"@commerce-storefront-toolset/core\\";
@NgModule({
    declarations: [],
    imports: [
        CheckoutRootModule,
        CheckoutB2BRootModule
    ],
    providers: [provideConfig(<CmsConfig>{
            featureModules: {
                [CHECKOUT_FEATURE]: {
                    module: () => import('./checkout-wrapper.module').then((m) => m.CheckoutWrapperModule),
                },
            }
        }),
        provideConfig(<I18nConfig>{
            i18n: {
                resources: checkoutTranslations,
                chunks: checkoutTranslationChunksConfig,
            },
        }),
        provideConfig(<I18nConfig>{
            i18n: {
                resources: checkoutB2BTranslations,
                chunks: checkoutB2BTranslationChunksConfig,
            },
        })
    ]
})
export class CheckoutFeatureModule {
}
"
`;

exports[`Spartacus Wrapper Module Schematics: ng g @commerce-storefront-toolset/schematics:wrapper-module One dynamic import in the file should generate appropriate feature module 2`] = `
"import { NgModule } from '@angular/core';
import { CheckoutB2BModule } from \\"@commerce-storefront-toolset/checkout/b2b\\";
import { CheckoutModule } from \\"@commerce-storefront-toolset/checkout/base\\";
@NgModule({
    declarations: [],
    imports: [
        CheckoutModule,
        CheckoutB2BModule
    ]
})
export class CheckoutWrapperModule {
}
"
`;

exports[`Spartacus Wrapper Module Schematics: ng g @commerce-storefront-toolset/schematics:wrapper-module wrapper module already exists should append the feature module after it, and not add a dynamic import to the feature module 1`] = `
"import { NgModule } from '@angular/core';
import { CheckoutModule } from \\"@commerce-storefront-toolset/checkout/base\\";
import { AnonymousConsentsModule, AuthModule, CostCenterOccModule, ExternalRoutesModule, ProductModule, ProductOccModule, UserModule, UserOccModule } from \\"@commerce-storefront-toolset/core\\";
import { DigitalPaymentsModule } from \\"@commerce-storefront-toolset/digital-payments\\";
import { AddressBookModule, AnonymousConsentManagementBannerModule, AnonymousConsentsDialogModule, BannerCarouselModule, BannerModule, BreadcrumbModule, CategoryNavigationModule, CmsParagraphModule, ConsentManagementModule, FooterNavigationModule, HamburgerMenuModule, HomePageEventModule, LinkModule, LoginRouteModule, LogoutModule, MyCouponsModule, MyInterestsModule, NavigationEventModule, NavigationModule, NotificationPreferenceModule, PageTitleModule, PaymentMethodsModule, ProductCarouselModule, ProductDetailsPageModule, ProductFacetNavigationModule, ProductImagesModule, ProductIntroModule, ProductListingPageModule, ProductListModule, ProductPageEventModule, ProductReferencesModule, ProductSummaryModule, ProductTabsModule, ScrollToTopModule, SearchBoxModule, SiteContextSelectorModule, StockNotificationModule, TabParagraphContainerModule } from \\"@commerce-storefront-toolset/storefront\\";
import { CartBaseFeatureModule } from './features/cart/cart-base-feature.module';
import { CheckoutFeatureModule } from './features/checkout/checkout-feature.module';
import { DigitalPaymentsFeatureModule } from './features/digital-payments/digital-payments-feature.module';
import { OrderFeatureModule } from './features/order/order-feature.module';
import { UserFeatureModule } from './features/user/user-feature.module';
@NgModule({
    declarations: [],
    imports: [
        AuthModule.forRoot(),
        LogoutModule,
        LoginRouteModule,
        HamburgerMenuModule,
        SiteContextSelectorModule,
        LinkModule,
        BannerModule,
        CmsParagraphModule,
        TabParagraphContainerModule,
        BannerCarouselModule,
        CategoryNavigationModule,
        NavigationModule,
        FooterNavigationModule,
        BreadcrumbModule,
        ScrollToTopModule,
        PageTitleModule,
        UserModule,
        UserOccModule,
        AddressBookModule,
        PaymentMethodsModule,
        NotificationPreferenceModule,
        MyInterestsModule,
        StockNotificationModule,
        ConsentManagementModule,
        MyCouponsModule,
        AnonymousConsentsModule.forRoot(),
        AnonymousConsentsDialogModule,
        AnonymousConsentManagementBannerModule,
        ProductModule.forRoot(),
        ProductOccModule,
        ProductDetailsPageModule,
        ProductListingPageModule,
        ProductListModule,
        SearchBoxModule,
        ProductFacetNavigationModule,
        ProductTabsModule,
        ProductCarouselModule,
        ProductReferencesModule,
        ProductImagesModule,
        ProductSummaryModule,
        ProductIntroModule,
        CostCenterOccModule,
        NavigationEventModule,
        HomePageEventModule,
        ProductPageEventModule,
        ExternalRoutesModule.forRoot(),
        UserFeatureModule,
        CartBaseFeatureModule,
        OrderFeatureModule,
        CheckoutFeatureModule,
        CheckoutModule,
        DigitalPaymentsFeatureModule,
        DigitalPaymentsModule
    ]
})
export class SpartacusFeaturesModule {
}
"
`;

exports[`Spartacus Wrapper Module Schematics: ng g @commerce-storefront-toolset/schematics:wrapper-module wrapper module already exists should append the feature module after it, and not add a dynamic import to the feature module 2`] = `
"import { NgModule } from '@angular/core';
import { checkoutTranslationChunksConfig, checkoutTranslations } from \\"@commerce-storefront-toolset/checkout/base/assets\\";
import { CheckoutRootModule } from \\"@commerce-storefront-toolset/checkout/base/root\\";
import { I18nConfig, provideConfig } from \\"@commerce-storefront-toolset/core\\";
@NgModule({
    declarations: [],
    imports: [
        CheckoutRootModule
    ],
    providers: [provideConfig(<I18nConfig>{
            i18n: {
                resources: checkoutTranslations,
                chunks: checkoutTranslationChunksConfig,
            },
        })
    ]
})
export class CheckoutFeatureModule {
}
"
`;
